frontend/packages/operator-lifecycle-manager/src/utils.ts

```diff
@@ -0,0 +1,34 @@
+import * as _ from 'lodash';
+import { ClusterServiceVersionKind } from './types';
+import { referenceForProvidedAPI } from './components';
+
+export const getInternalObjects = (csv: ClusterServiceVersionKind) => {
+  const internals: string = _.get(csv, [
+    'metadata',
+    'annotations',
+    'operators.operatorframework.io/internal-objects',
+  ]);
+  if (!internals) {
+    return [];
+  }
+  try {
+    return JSON.parse(internals);
+  } catch (e) {
+    // eslint-disable-next-line no-console
+    console.error('Error parsing internal object annotation.', e);
+    return [];
+  }
+};
+
+export const isInternalObject = (internalObjects: string[], objectName: string): boolean =>
+  internalObjects.some((obj) => obj === objectName);
+
+export const getInternalAPIReferences = (csv: ClusterServiceVersionKind): string[] => {
+  const { owned } = csv.spec.customresourcedefinitions;
```

Comment:
I'm not sure this is required, so this could result in a runtime error. Better to guard against that

```suggestion
  const owned: CRDDescription[] = _.get(csv, 'spec.customresourcedefinitions') || [];
```